locations.js

```diff
@@ -0,0 +1,252 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+const Gtk = imports.gi.Gtk;
+const Shell = imports.gi.Shell;
+const Signals = imports.signals;
+
+// Use __ () and N__() for the extension gettext domain, and reuse
+// the shell domain with the default _() and N_()
+const Gettext = imports.gettext.domain('dashtodock');
+const __ = Gettext.gettext;
+const N__ = function(e) { return e };
+
+const Me = imports.misc.extensionUtils.getCurrentExtension();
+const Utils = Me.imports.utils;
+
+/**
+ * This class maintains a Shell.App representing the Trash and keeps it
+ * up-to-date as the trash fills and is emptied over time.
+ */
+var Trash = class DashToDock_Trash {
+
+    constructor() {
+        this._file = Gio.file_new_for_uri('trash://');
+        this._monitor = this._file.monitor_directory(0, null);
```

Comment:
File operations can fail, causing the whole dash stop working when some permissions aren't set, so should use more `try/catch` here

Reply:
Fixed with your suggestion.